docs/docsite/rst/plugins/inventory.rst

```diff
@@ -80,12 +86,24 @@ You can create dynamic groups using host variables with the constructed ``keyed_
       - key: tags.Name
         prefix: tag_Name_
         separator: ""
+      # If you have a tag called "Role" which has the value "Webserver", this will add the group
+      # Role_Webserver and add any hosts that have that tag assigned to it.
+      - key: tags.Role
+        prefix: role
     groups:
       # add hosts to the group development if any of the dictionary's keys or values is the word 'devel'
       development: "'devel' in (tags|list)"
+      # add hosts to the "private_only" group if the host doesn't have a public IP associated to it
+      private_only: "(network_interfaces[0].association | default({}) ).public_ip | default('') != ''"
```

Comment:
As long as public_ip isn't sometimes an empty string, this could be
```suggestion
      private_only: "(network_interfaces[0].association | default({}) ).public_ip is defined
```

Why not check the top level variable `public_ip_address` though?

Reply:
Ah, so let's cover off the "why not check the top level variable" - because I completely missed it! :) I looked for it on a node that didn't have a public IP, and it wasn't there, so I figured I'd have to "find it" somewhere else, and then went looking for where the attachment happened elsewhere - `network_interfaces`. Having now looked for it, I can clearly see it. D'oh.

Also yes, `is defined` versus `| default('') != ''` makes much more sense. I have hit issues with this in the dim and distant past, mostly with parsing environment variables, and so my "default reaction" is to go with the latter form rather than the former. Yes, I need to fix this one!

Reply:
Oh, an interesting quirk. `public_ip_address` is only defined where the first interface has a public ip defined. If you attach an EIP to the second or later interfaces, then it's not classed as that host's public ip address. This is totally outside the scope of this documentation change, but it gives me a bit more context for why I might have missed this before :)